### PR TITLE
Add magnetometer support to Android via Input.get_magnetometer()

### PR DESCRIPTION
**Pull request by ghost**
Someone please review.

### CODE REVIEW

**Comment by punto-** [comment]
Is there a case where the computer might have more than 1 magnetometer?
Like on a PC or a console where you attach peripherals?

**Comment by fluffrabbit** [comment]
Loving the new smilies.

Hmm. On mobile devices, no. Even if an OTG magnetometer were plugged in, Android and probably other platforms will favor their one internal magnetometer for instance in orientation calculations since it reflects the angle of the device. This works the exact same way as Input.get_accelerometer() (a lot of copying and pasting). For PCs and such, this is an unknown. Maybe there are controllers with accelerometers and magnetometers (PS3 controller or something) and if multiple controllers were plugged in then there would be multiple input sources, but AFAIK Input.get_accelerometer() doesn't account for this possibility as it only applies to internal mobile accelerometers (and now magnetometers). Maybe some functions could be renamed or whatever and I can't speak for all platforms but I know that the way Android is set up it really likes to pretend that its own internal magnetometer (if available) is the only one in the world.

File: core/os/input.h [hunk 1]
```diff
@@ -55,7 +55,7 @@ class Input : public Object {
 
 	static Input *get_singleton();
 
-	virtual bool is_key_pressed(int p_scancode)=0;
+	virtual bool is_key_pressed(int p_scancode)=0;
```
Comment (est31) [inline core/os/input.h]: This is a trailing whitespace, and should be avoided. Can you remove it?

File: platform/android/os_android.h [hunk 2]
```diff
@@ -193,9 +193,9 @@ class OS_Android : public OS_Unix {
 	virtual void set_video_mode(const VideoMode& p_video_mode,int p_screen=0);
 	virtual VideoMode get_video_mode(int p_screen=0) const;
 	virtual void get_fullscreen_mode_list(List<VideoMode> *p_list,int p_screen=0) const;
-
+
```
Comment (est31) [inline platform/android/os_android.h]: trailing whitespace

**Comment by fluffrabbit** [comment]
Removed reported whitepace and removed comment flags of my name as discussed on IRC.

**Comment by akien-mga** [comment]
> fixed some line ending bullshit for @tomreyn

Really? Based on that and tonight's logs (http://godot.eska.me/irc-logs/devel/2016-03-12.log), I think you really need to review your attitude. We don't merge code from people that don't respect the rest of the community. If it's a wanted feature, I very much welcome another contributor to cherry-pick the changes, merge them all into one and remove offensive comments from the commit logs.

**Pull request closed without merging**